File: tabulate-core/src/test/java/io/github/voytech/tabulate/core/builder/fluent/support/TestResultProvider.java
package io.github.voytech.tabulate.core.builder.fluent.support;

import io.github.voytech.tabulate.template.result.ResultProvider;
import kotlin.Unit;
import org.jetbrains.annotations.NotNull;

public class TestResultProvider implements ResultProvider<TestRenderingContext, Unit> {
    @NotNull
    @Override
    public Class<Unit> outputClass() {
        return Unit.class;
    }

    @Override
    public void setOutput(@NotNull TestRenderingContext renderingContext, Unit output) {

    }

    @Override
    public void flush() {

    }
}

File: tabulate-core/src/test/kotlin/io/github/voytech/tabulate/support/TestResultProvider.kt
package io.github.voytech.tabulate.support

import io.github.voytech.tabulate.template.result.ResultProvider
import java.util.logging.Logger

class TestResultProvider: ResultProvider<TestRenderingContext, Unit> {

    override fun outputClass() = Unit.javaClass

    override fun flush() {
        logger.info("This is fake implementation of ResultProvider flushing results into ether")
    }

    override fun setOutput(renderingContext: TestRenderingContext, output: Unit) {
        logger.info("This is fake implementation of ResultProvider flushing results into ether")
    }

    companion object {
        val logger: Logger = Logger.getLogger(TestResultProvider::class.java.name)
    }
}

class AlternativeTestResultProvider: ResultProvider<AlternativeTestRenderingContext, Unit> {

    override fun outputClass() = Unit.javaClass

    override fun flush() {
        logger.info("This is fake implementation of ResultProvider flushing results into ether")
    }

    override fun setOutput(renderingContext: AlternativeTestRenderingContext, output: Unit) {
        logger.info("This is fake implementation of ResultProvider flushing results into ether")
    }

    companion object {
        val logger: Logger = Logger.getLogger(TestResultProvider::class.java.name)
    }
}